pilot/pkg/model/gateway.go

```diff
@@ -253,6 +265,23 @@ func MergeGateways(gateways []gatewayWithInstances, proxy *Proxy) *MergedGateway
 						} else {
 							mergedServers[serverPort].Servers = append(mergedServers[serverPort].Servers, s)
 						}
+
+						// We have TLS settings defined and we have already taken care of unique route names
+						// if it is HTTPS. So we can construct a QUIC server on the same port. It is okay as
+						// QUIC listens on UDP port, not TCP
+						if features.EnableQUICListeners && gateway.IsEligibleForHTTP3Upgrade(s) &&
+							udpSupportedPort(s.GetPort().GetNumber(), gwAndInstance.instances) {
+							log.Debugf("Server at port %d eligible for HTTP3 upgrade. Add UDP listener for QUIC", serverPort.Number)
+							if mergedQUICServers[serverPort] == nil {
+								mergedQUICServers[serverPort] = &MergedServers{
```

Comment:
very nit: prefer setting it to empty MergedServers then do not have an `else` - always set it. That way its consistent code if we need a change

Reply:
Done